### ISSUE TITLE
API multi select integer key not working

### ISSUE THREAD

**Issue by Jojolam44** [AdvancedCustomFields/acf#669]
Hi,

We have a problem with select field in API POST.

In the app configuration, we have these parameters (integer keys) :
![image](https://user-images.githubusercontent.com/4331282/175961127-8515b40a-96e2-4185-a0aa-0b495356c73f.png)

But the API required string and no integers. When we POST values in string, WordPress can't find the value in the enum.
The problem only appear with the multivalued select (array), it's OK when we post single value in string.

I've temporarily fixed this problem by modifying wp-rest.api:1935 (rest_are_values_equal function) : "==" instead of "===".

How can we fix this ?

Thanks for your answers.

**Comment by mattgrshaw** [comment]
Hi @Jojolam44,

Thanks for the report - I was able to replicate this one as well. It looks like we'll likely have to update the REST schema for the multiselect to allow integers as well. I've raised an issue on our end so we can look into this for a future update.

I was able to get this working by editing the ACF `includes/fields/class-acf-field-select.php` file to update the schema array starting on line [691](https://github.com/AdvancedCustomFields/acf/blob/master/includes/fields/class-acf-field-select.php#L691) so that it looks like this:

	$schema = array(
		'type'     => array( 'string', 'int', 'array', 'null' ),
		'required' => ! empty( $field['required'] ),
		'items'    => array(
			'type' => array( 'string', 'int' ),
			'enum' => empty( $option_keys ) ? $field['choices'] : $option_keys,
		),
	);

I generally don't recommend editing plugin files, but I think in this case it's better than modifying WordPress core files and potentially altering API requests/responses for non-ACF API calls.
